src/jit/lir.cpp

```diff
@@ -195,7 +195,14 @@ void LIR::Use::ReplaceWith(Compiler* compiler, GenTree* replacement)
     *m_edge = replacement;
     if (!IsDummyUse() && m_user->IsCall())
     {
-        compiler->fgFixupArgTabEntryPtr(m_user, replacedNode, replacement);
+        GenTreeCall* call = m_user->AsCall();
+        const bool isCallArg = (replacement != call->gtControlExpr) && ((call->gtCallType != CT_INDIRECT) ||
```

Comment:
I don't think it would be a good idea to use `gtGetArgEntryByNode` - it is unnecessarily expensive.
Instead, one might add something like an `IsNonArgOperand()` to `GenTreeCall` that would check for these special cases - that way the checking would be with the declaration of the fields of that node.

---
In reply to: [78644811](https://github.com/dotnet/coreclr/pull/7170#discussion_r78644811) [](ancestors = 78644811)